### PR TITLE
Split pulseio.PWMOut into pwmio

### LINKED ISSUE #3211: Move `PWMOut` to new `pwmio` module
In, 6.0.0 alias it in. Once that is done, we'll remove it from `pulseio` in 7.0.0.

### PR DESCRIPTION
**Pull request by tannewt**
This gives us better granularity when implementing new ports because
PWMOut is commonly implemented before PulseIn and PulseOut.

Fixes #3211

### CODE REVIEW

**Comment by tannewt** [comment]
@dhalbert We should be able to do beta releases after this.

File: ports/esp32s2/common-hal/pwmio/__init__.c [hunk 1]
```diff
@@ -0,0 +1 @@
+// No pulseio module functions.
```
Comment (hierophect) [inline ports/esp32s2/common-hal/pwmio/__init__.c:1]: wrong name

File: ports/cxd56/common-hal/pwmio/__init__.c [hunk 2]
```diff
@@ -0,0 +1 @@
+// No pulseio module functions.
```
Comment (hierophect) [inline ports/cxd56/common-hal/pwmio/__init__.c:1]: accidental wrong name here?

File: ports/mimxrt10xx/common-hal/pwmio/__init__.c [hunk 3]
```diff
@@ -0,0 +1 @@
+// No pulseio module functions.
```
Comment (hierophect) [inline ports/mimxrt10xx/common-hal/pwmio/__init__.c:1]: wrong name

**Comment by tannewt** [comment]
Ok, everything is passing now. Please take another look.

File: ports/cxd56/common-hal/pulseio/PulseOut.c [hunk 4]
```diff
@@ -59,7 +59,7 @@ static bool pulseout_timer_handler(unsigned int *next_interval_us, void *arg)
 }
 
 void common_hal_pulseio_pulseout_construct(pulseio_pulseout_obj_t* self,
-                                            const pulseio_pwmout_obj_t* carrier,
+                                            const pwmio_pwmout_obj_t* carrier,
```
Comment (hierophect) [inline ports/cxd56/common-hal/pulseio/PulseOut.c:62]: This file is different than the other PulseOut.c files in common-hal, in that it appears to access the pulseio file include through `"shared-bindings/pulseio/PulseOut.h"`. Should we make that consistent across all files, or add in a direct reference to pulseio in the spirit of include what you use?

Could be revisited as part of a more general inclusion style review.

Comment (tannewt) [reply ports/cxd56/common-hal/pulseio/PulseOut.c:62]: Ya, generally it's better to include what you use but ensuring that in a manual way is impossible. I think this is fine for now.

**Comment by hierophect** [comment]
Looks good, thanks!

**Pull request merged**